runtime/src/lib.rs

```diff
@@ -205,7 +228,21 @@ impl<'a> Drivers<'a> {
     }
 }
 
-fn wait_for_cmd(_mbox: &mut Mailbox) {
+
+/// Run pending jobs and enter low power mode.
+fn goto_idle(drivers: &mut Drivers) {
+
+    // Run pending jobs before entering low power mode.
+    #[cfg(feature = "fips_self_test")]
+    if let SelfTestStatus::InProgress(execute) = drivers.self_test_status {
+        if drivers.mbox.lock() == false {
+            match execute(drivers) {
+                Ok(_) => drivers.self_test_status = SelfTestStatus::Done,
+                Err(e) => caliptra_drivers::report_fw_error_non_fatal(e.into()),
+            }
+        }
+    }
```

Comment:
Is there a specific reason SelfTest can't be blocking?

Reply:
We need to  “reply with data” to conform with new FIPS API (response contains header + checksum), but the mailbox SRAM would already be mutated with the image.

Reply:
The alternative approach is to make the image-verification digest generic, using the sha384 peripheral instead of sha384acc with the mailbox.

If we want to abandon calling back into ROM forever, perhaps just using the sha384 peripheral is better...

Reply:
I think we should keep this run-self-test mailbox API as is. If a future version of FIPS ever needs us to test the sha384acc peripheral with something other than empty input data we'll need a mechanism to modify the mailbox.